### PR TITLE
added better support for variable alias when transforming tokens using style dictionary

### PR DESCRIPTION
**Pull request by ryanzec**
This makes changes so that the outputted design token can properly be transformed with style dictionary when using aliases for a variable value.

This PR adds in functionality so that after all variables are processed, it does a second pass on the variables and if the variable is an alias, it will basically create a copy of the variable and update both the `name` and the `values` of the variable to account for the modes of the alias.

Previous to these changes, this is an example of a alias variables design token output:

Screen Shot 2023-08-06 at 11 32 23 AM
Screen Shot 2023-08-06 at 11 32 13 AM

With the changes, the output is now:

Screen Shot 2023-08-06 at 11 34 19 AM
Screen Shot 2023-08-06 at 11 34 11 AM
Screen Shot 2023-08-06 at 11 34 05 AM

Adding for the mode in the `values` is needed to make sure when transforming with style dictionary, the reference can be found. Adding the mode in the `name` is needed to make sure there are not duplicate variables names.

This seems like the cleanest way to add this functionality but let me know if there is any feedback.

This addresses #260

### CODE REVIEW

**Comment by lukasoppermann** [comment]
Hey, I think this is certainly one option, as I am also not sure exporting multiple files works. (Haven't had time to look into it).

I would like this feature to be togglable via the UI. I guess we could tie this the `Reference mode in variables` option?

If it is one, than it would reference modes in both the file structure on references, if it is off, it will be removed from both. WDYT?

**Comment by ryanzec** [comment]
I think that makes sense since when mode reference is on and aliasing a variable with multiple modes does not work right now when transforming anyways.

@lukasoppermann I believe I have updated the PR based on your feedback but if I did not understand it correctly, let me know.

**Comment by ryanzec** [comment]
@lukasoppermann Is there anything else this PR needs?

File: src/utilities/getVariables.ts [hunk 1]
```diff
@@ -55,6 +61,38 @@ const extractVariable = (variable, value) => {
   }
 }
 
+const processAliasModes = (variables) => {
+  return variables.reduce((collector, variable) => {
+    // nothing needs to be done to variables that has no alias modes
```
Comment (lukasoppermann) [inline src/utilities/getVariables.ts:66]:
```suggestion
    // nothing needs to be done to variables that have no alias modes
```

File: src/utilities/getVariables.ts [hunk 2]
```diff
@@ -55,6 +61,38 @@ const extractVariable = (variable, value) => {
...
+  return variables.reduce((collector, variable) => {
+    // nothing needs to be done to variables that have no alias modes
+    if (!variable.aliasModes) {
+      collector.push(variable)
+
+      return collector
+    }
+
+    const aliasModes = variable.aliasModes
+    const aliasName = variable.aliasName
```
Comment (lukasoppermann) [inline src/utilities/getVariables.ts:74]:
```suggestion
    const { aliasModes, aliasName } = variable
```

**Comment by lukasoppermann** [comment]
Ahh, @ryanzec so the issue is, that you are replacing the reference and the variable name with the same mode. however the referenced variable is different and may come from a different mode and collection.

**Comment by lukasoppermann** [comment]
Hey @ryanzec I had to rollback this PR. I didn't find the time to fix it and its blocking the merge of other changes.

I tried to fix it in here: https://github.com/lukasoppermann/design-tokens/tree/fixForModesExport but did not succeed yet.

Can you send a new PR with this and try to fix it.

Use cases it needs to cover:
- reference without modes
- normal tokens
- reference with multiple modes
- modeReference set to false

**Comment by hayawata3626** [comment]
@lukasoppermann 
Which items are currently not being met?
If I have a time, I will debug.

> reference without modes
> normal tokens
> reference with multiple modes
> modeReference set to false

**Comment by hayawata3626** [comment]
@lukasoppermann 
I tried to fix the problem and pushed the codes.
If you have a time, please check.
https://github.com/hayawata3626/design-tokens/tree/fix-get-variables

**Pull request merged**